### PR TITLE
Misc autotools fixes

### PR DESCRIPTION
**Pull request by NattyNarwhal**
Fixes the build on AIX and FreeBSD, and some stuff for Haiku. (More Haiku stuff is blocking on upstream CoreFX merging the Haiku PAL patchset.)

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -3019,17 +3023,9 @@ if test x$host_win32 = xno; then
...
-			#include <fcntl.h>
-		], [
-			struct flock64;
-		],[
-			AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_FLOCK64, 1, struct flock64)
-		], [
-			AC_MSG_RESULT(no)
-		])
+		# We can't match autotools with structs, so try using a member we know will
```
Comment (jaykrell) [inline configure.ac]: I don't understand the comment.

Comment (NattyNarwhal) [reply configure.ac]: comment dated back to an older patch that matched on a structure member, it's senseless now that it matches on type and I'll remove it

**Comment by jaykrell** [comment]
Thank you/sorry. I do appreciate lack of incorrect comments, and optionally correct comments. :)

**Comment by luhenry** [comment]
@monojenkins build

**Pull request merged**